specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx

```diff
@@ -274,7 +274,12 @@ export function QueryResults(props: QueryResultsProps): JSX.Element {
           overflow-auto rounded
           ${tableClassName}
           ${showResults ? 'border-b border-gray-500' : ''}
-       `}
+          ${
```

Comment:
not a correct solution (not commenting on whether it works or not)
can you spot the issue? look closely at some of the other classes AND props on this div

Reply:
I forgot to remove the other class name that contained already the grid but I think the solution I have is the one that works. Any other suggestion?

Reply:
thanks for the clarification

a few comments:
- the code no longer handles the `showLineNumber` case (see the `--meta-columns` CSS variable declared a ew lines above, and that variable was used in the class name before) resulting in a bug in cases when that pref is set
- given that `--meta-columns` is no longer used, it should be removed

it seems like the real issue is that repeat() CSS function does not allow `0` as the first value - https://developer.mozilla.org/en-US/docs/Web/CSS/repeat#:~:text=It%20is%20specified%20with%20an%20integer%20value%20of%201%20or%20more
and when that value happens to be 0, it breaks the grid-cols, causing this issue

thus, I would instead suggest you to add a style like this (rather than a tailwind class name as it's getting too complex for tailwind):

```ts
const metaColumns = (showLineNumber ? 1 : 0) + (hasIdField ? 2 : 0);

// later:
style={{
  gridTemplateColumns: `${metaColumns === 0 ? '' : `repeat(${metaColumns}, min-content) `}repeat(${visibleFieldSpecs.length}, auto)`;
}}
```

or alternative version (using js fill rather than CSS repeat):

```ts
const metaColumns = (showLineNumber ? 1 : 0) + (hasIdField ? 2 : 0);
style={{
  gridTemplateColumns: [...Array.from({length:metaColumns}).fill('min-content'), ...Array.from(visibleFieldSpecs).fill('min-content')].join(', ');
}}
```

and then --columns and --meta-columns CSS variable should be removed